src/control/common/file_utils.go

```diff
@@ -46,13 +46,43 @@ const (
 	UtilLogDepth = 4
 )
 
-// GetAbsInstallPath retrieves absolute path of files in daos install dir
-func GetAbsInstallPath(relPath string) (string, error) {
+// CurrentExecPath retrieves absolute path of this executable.
+func CurrentExecPath() (string, error) {
 	ex, err := os.Executable()
 	if err != nil {
 		return "", err
 	}
-	return filepath.Join(filepath.Dir(ex), "..", relPath), nil
+	return filepath.Dir(ex), nil
+}
+
+// GetAbsPath retrieves absolute from relative path.
+func GetAbsPath(path string) (string, error) {
```

Comment:
This function seems to be misleadingly named... It's returning an absolute path if the path was already absolute, otherwise it's "absolute" relative to the binary? If it were an unexported utility function it wouldn't bother me so much, but this is our internal API surface, and it shouldn't be necessary to read the function to figure out what it does.

What would be clearest, if slightly less less convenient, would be to do the filepath.IsAbs() check separately and only call this function if the path is not absolute. Perhaps it could be called ResolveAdjacentPath() ?

Reply:
I'm not sure what you mean, it always returns an absolute path starting with "/". if the path given is not absolute then the absolute path of the running executable will be prepended to the supplied relative path to give the absolute path of the relative input path from the perspective of the running executables location. I think it's fairly clear but maybe I'm missing something.

Reply:
I guess what I'm reacting to is the fact that a user of this function would have to read the function body to understand what it does. You know what it does because you wrote it. I know what it does because I went and studied the code. But if I, as a new control plane developer or other unfamiliar reader were to look at the comment and the name, then taken at face value it indicates absolutely nothing about the fact that the path to running executable will be resolved as part of constructing an absolute path to the specified file. 

Again, if this were an unexported convenience function defined near the code where it's used I wouldn't really care, although I'd still think the name is not particularly helpful.

It's not meant to be an insult -- I'm just advocating for our internal documentation and package APIs to be as clear and useful as what we might export to 3rd-party developers.